src/javaharness/java/arcs/api/CrdtCollectionModel.java

```diff
@@ -0,0 +1,140 @@
+package arcs.api;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class CrdtCollectionModel {
+    public static class ModelValue {
+        public final String id;
+        public final PortableJson rawData;
+        public final String storageKey;
+
+        ModelValue(String id, PortableJson rawData, String storageKey) {
+            this.id = id;
+            this.rawData = rawData;
+            this.storageKey = storageKey;
+        }
+    }
+
+    public static class SerializedModelEntry {
+        public final String id;
+        public final ModelValue value;
+        public final List<String> keys;
+
+        SerializedModelEntry(String id, ModelValue value, List<String> keys) {
+            this.id = id;
+            this.value = value;
+            this.keys = keys;
+        }
+    }
+    public static class ModelEntry {
+        public ModelValue value;
+        public final Set<String> keys;
+        ModelEntry(ModelValue value, List<String> keys) {
+            this.value = value;
+            this.keys = new HashSet();
+            if (keys != null) {
+                for (String key: keys) {
+                    this.keys.add(key);
+                }
```

Comment:
`this.keys.addAll(keys);`

Reply:
I talked to Shane about it and he recommended I translate the new crdt classes, instead of the old ones, so I completely removed this file from this PR.
but thanks for the comments (I used to know Java better than this, hopefully, it'll all come back to me :)